### ISSUE TITLE
FStar.IndefiniteDescription.fst fails to parse with fsdoc. 

### ISSUE THREAD

**Issue by sishtiaq** [FStarLang/FStar#761]
sishtiaq@SaminDesktop MINGW64 /e/FStar/ulib (sishtiaq_fsdoc1)
$ fstar --doc FStar.IndefiniteDescription.fst
.\FStar.IndefiniteDescription.fst(7,13-7,14) : (Error) Syntax error

**Comment by catalin-hritcu** [comment]
nik:
you must be using the F# version of F*
that has a known but very mysterious parsing bug
that causes it to not parse stuff like this:   (x:a & p x)

catalin:
it works indeed with the OCaml version, fails only on F#
@sishtiaq closing, this is a duplicate of #404

**Comment by catalin-hritcu** [comment]
We should really try to fix the lexing bugs in F#  (#404 and #252)

Still, the workaround for these bugs does work here too:
```
module FStar.IndefiniteDescription

type bar (i:int) = int // working around #404

// This is the mother of a lot of axioms, use with care!

// TODO:Type0 should be prop
assume val indefinite_description : a:Type -> p:(a->GTot Type0) -> Ghost
  (x:a & p x)
  (requires (exists x. p x))
  (ensures (fun _ -> True))
```
